lib/src/iterables/merge.dart

```diff
@@ -89,5 +93,8 @@ class _MergeIterator<T> implements Iterator<T> {
   }
 
   @override
-  T get current => _current;
+  T get current {
+    final cur = _current;
+    return cur as T;
```

Comment:
same here: too trivial to warrant a local variable

Reply:
Done.